### DISCUSSION TITLE
Example of useForm with mutation using different resource, than what's used to fetch the data?

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2562] [Q&A] [answered]
### Problem Space

Most API's often include `Internal/Hidden` routes inaccessible in the public API schema 

A REST endpoint used to query the data often differs from the endpoint used to patch/delete/post data. (While get routes can be also hidden from public API)

Let's say I need to fetch a user at Get: `Public/User/{id}`, but to modify a user I need to use  Patch: `Internal/User/{id}`

So you give a resource to useForm:
```tsx
	const {
		formProps,
		mutationResult,
		show
	} = useForm<TData, TError, TUpdates>({
		action: 'edit',
		resource: `Public/User/${showId as string}`,
                // mutation needs to happen at Internal/User/${showId as string}
		queryOptions: {
			enabled: Boolean(showId)
		},
	})
```

But then useMutation uses the same route to `PATCH` the data as useQuery, which I see as an antipattern if u can't override a given mutation from formProps with ur own, or explicitly tell it to use a different API req. Or even an advice.

**Comment by omeraplak** [comment, marked as answer]
Hey @vinclou ,
I don't know how common this use-case is but you can deal with this using data hooks. [Override your form's onSubmit ](https://refine.dev/docs/faq/#how-can-i-change-the-form-data-before-submitting-it-to-the-api) and call your desired data hook (`useUpdate`, `useCreate`, etc.). For data store invalidation, you can use [useInvalidate](https://refine.dev/docs/api-reference/core/hooks/invalidate/useInvalidate/). I hope it will be useful for you 💥

**Reply by vinclou** [reply]
True, that's how I solved the problem yesterday, I was just double checking if I'm missing anything
